RecoPixelVertexing/PixelVertexFinding/plugins/PixelVertexCollectionTrimmer.cc

```diff
@@ -1,128 +1,100 @@
 #include "FWCore/Framework/interface/MakerMacros.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
-
-#include "DataFormats/VertexReco/interface/Vertex.h"
-#include "DataFormats/VertexReco/interface/VertexFwd.h"
-#include "FWCore/Utilities/interface/InputTag.h"
-#include "RecoPixelVertexing/PixelVertexFinding/interface/PVClusterComparer.h"
-
 #include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
 #include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "RecoPixelVertexing/PixelVertexFinding/interface/PVClusterComparer.h"
 
 class PixelVertexCollectionTrimmer : public edm::stream::EDProducer<> {
 public:
   explicit PixelVertexCollectionTrimmer(const edm::ParameterSet&);
-  ~PixelVertexCollectionTrimmer() override;
 
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
 
 private:
   void produce(edm::Event&, const edm::EventSetup&) override;
 
-  edm::EDGetTokenT<reco::VertexCollection> vtxToken_;
-  unsigned int maxVtx_;
-  double fractionSumPt2_;
-  double minSumPt2_;
+  edm::EDGetTokenT<reco::VertexCollection> const vtxToken_;
+  uint const maxVtx_;
+  double const fractionSumPt2_;
+  double const minSumPt2_;
 
-  PVClusterComparer* pvComparer_;
+  std::unique_ptr<PVClusterComparer> pvComparer_;
 };
 
-PixelVertexCollectionTrimmer::PixelVertexCollectionTrimmer(const edm::ParameterSet& iConfig) {
-  edm::InputTag vtxInputTag = iConfig.getParameter<edm::InputTag>("src");
-  vtxToken_ = consumes<reco::VertexCollection>(vtxInputTag);
-  maxVtx_ = iConfig.getParameter<unsigned int>("maxVtx");
-  fractionSumPt2_ = iConfig.getParameter<double>("fractionSumPt2");
-  minSumPt2_ = iConfig.getParameter<double>("minSumPt2");
-
-  edm::ParameterSet PVcomparerPSet = iConfig.getParameter<edm::ParameterSet>("PVcomparer");
-  double track_pt_min = PVcomparerPSet.getParameter<double>("track_pt_min");
-  double track_pt_max = PVcomparerPSet.getParameter<double>("track_pt_max");
-  double track_chi2_max = PVcomparerPSet.getParameter<double>("track_chi2_max");
-  double track_prob_min = PVcomparerPSet.getParameter<double>("track_prob_min");
-
-  pvComparer_ = new PVClusterComparer(track_pt_min, track_pt_max, track_chi2_max, track_prob_min);
+PixelVertexCollectionTrimmer::PixelVertexCollectionTrimmer(const edm::ParameterSet& iConfig)
+    : vtxToken_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("src"))),
+      maxVtx_(iConfig.getParameter<uint>("maxVtx")),
+      fractionSumPt2_(iConfig.getParameter<double>("fractionSumPt2")),
+      minSumPt2_(iConfig.getParameter<double>("minSumPt2")) {
+  if (fractionSumPt2_ < 0)
+    edm::LogWarning("Configuration") << "value of \"fractionSumPt2\" is negative.";
+  else if (fractionSumPt2_ > 1)
+    edm::LogWarning("Configuration") << "value of \"fractionSumPt2\" is larger than 1.";
+
+  auto const PVcomparerPSet = iConfig.getParameter<edm::ParameterSet>("PVcomparer");
```

Comment:
```suggestion
  auto const& pvComparerPSet = iConfig.getParameterSet("PVcomparer");
```
- capitalization
- using a method returning by ref may be cheaper

Reply:
Done.